### PR TITLE
Feature/add claim

### PR DESCRIPTION
**Pull request by Jedi2002**

### CODE REVIEW

File: x/dispensation/handler_test.go [hunk 1]
```diff
@@ -32,3 +33,19 @@ func TestNewHandler(t *testing.T) {
...
+
+func TestNewHandler_CreateClaim(t *testing.T) {
+	app, ctx := test.CreateTestApp(false)
+	keeper := app.DispensationKeeper
+	handler := dispensation.NewHandler(keeper)
+	address := sdk.AccAddress(crypto.AddressHash([]byte("User1")))
+	msgClaim := types.NewMsgCreateClaim(address, types.ValidatorSubsidy)
+	res, err := handler(ctx, msgClaim)
+	require.NoError(t, err)
+	require.NotNil(t, res)
```
Comment (timlind) [inline x/dispensation/handler_test.go:45]: Are there other assertions that can be put here?

File: x/dispensation/keeper/distrubtions_test.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+func TestKeeper_GetDistributions(t *testing.T) {
+	app, ctx := test.CreateTestApp(false)
+	keeper := app.DispensationKeeper
+	for i := 0; i < 10; i++ {
+		name := uuid.New().String()
+		distribution := types.NewDistribution(types.Airdrop, name)
+		err := keeper.SetDistribution(ctx, distribution)
+		assert.NoError(t, err)
+		_, err = keeper.GetDistribution(ctx, name, types.Airdrop)
```
Comment (timlind) [inline x/dispensation/keeper/distrubtions_test.go:19]: Are there other assertion to put on distribution res?

**Review by timlind — DISMISSED** [review]

File: x/dispensation/types/msgs.go [hunk 3]
```diff
@@ -28,6 +28,7 @@ func (m MsgDistribution) Route() string {
 	return RouterKey
 }
 
+//TODO Replace with constant defined in keys.go with value CreateDispensation
```
Comment (iverc) [inline x/dispensation/types/msgs.go:31]: Should we address it in this PR?

File: x/dispensation/types/querier.go [hunk 4]
```diff
@@ -8,14 +8,15 @@ const (
 	QueryAllDistributions   = "distributions"
 	QueryRecordsByDistrName = "records_by_name"
 	QueryRecordsByRecipient = "records_by_recipient"
+	QueryClaimsByType       = "claims_by_type"
 )
 
 type QueryRecordsByDistributionName struct {
```
Comment (iverc) [inline x/dispensation/types/querier.go:14]: I believe we should define types in proto now.

File: x/dispensation/types/records.go [hunk 5]
```diff
@@ -25,19 +26,20 @@ func (d ClaimStatus) String() string {
 }
 
 // DistributionRecord is created for every recipient for a distribution
-// TODO : Add ClaimStatus to the prefixed key for records
+// TODO : Add DistributionStatus to the prefixed key for records
```
Comment (iverc) [inline x/dispensation/types/records.go:29]: Should it be addressed as part of this PR?

File: x/dispensation/types/querier.go (hunk 4 above)
Comment (timlind) [reply x/dispensation/types/querier.go:14]: This may already be done in the gRPC PR

File: x/dispensation/keeper/distrubtions_test.go (hunk 2 above)
Comment (Jedi2002) [reply x/dispensation/keeper/distrubtions_test.go:19]: I added the test for GetDistributions and it has a check for the total list .

File: x/dispensation/types/msgs.go (hunk 3 above)
Comment (Jedi2002) [reply x/dispensation/types/msgs.go:31]: No , this is more of a cosmetic change , I put that there to keep track . Just trying to keep changes minimal right now , and add everything in bulk witht he .42 upgrade.

File: x/dispensation/types/records.go (hunk 5 above)
Comment (Jedi2002) [reply x/dispensation/types/records.go:29]: Same as above , ill make these changes in the .42 version

File: x/dispensation/handler_test.go (hunk 1 above)
Comment (Jedi2002) [reply x/dispensation/handler_test.go:45]: We are checking the using GetClaim below , so the functionality works ,the claims get saved . 
Checking the hash and verifying it by querying the blockchain should come under the integration tests I guess.

**Review by marshsif — APPROVED** [review]

**Pull request merged**